src/main.cc

```diff
@@ -1860,13 +1860,12 @@ GoIntoBackground()
     pid_t pid;
     if ((pid = fork()) < 0) {
         int xerrno = errno;
-        syslog(LOG_ALERT, "fork failed: %s", xstrerr(xerrno));
-        // continue anyway, mimicking --foreground mode (XXX?)
+        fatalf("fork failed: %s", xstrerr(xerrno));
```

Comment:
If possible, please throw an exception instead. Modern Squids use C++ exceptions for error reporting/handling (where those exceptions work and are appropriate). This is probably one of those works-and-appropriate cases.
```suggestion
        throw TexcHere(ToSBuf("failed to fork(2) the master process: ", xstrerr(xerrno)));
```